internal/builtin/data_http.go

```diff
@@ -0,0 +1,271 @@
+package builtin
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"log/slog"
+	"mime"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+	"unicode/utf8"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/blackstork-io/fabric/internal/builtin/utils"
+	"github.com/blackstork-io/fabric/pkg/diagnostics"
+	"github.com/blackstork-io/fabric/plugin"
+	"github.com/blackstork-io/fabric/plugin/dataspec"
+	"github.com/blackstork-io/fabric/plugin/dataspec/constraint"
+)
+
+func makeHTTPDataSource(version string) *plugin.DataSource {
+	return &plugin.DataSource{
+		DataFunc: fetchHTTPDataWrapper(version),
+		Args: dataspec.ObjectSpec{
+			&dataspec.AttrSpec{
+				Name:        "url",
+				Type:        cty.String,
+				ExampleVal:  cty.StringVal("https://example.localhost/file.json"),
+				Constraints: constraint.RequiredMeaningful,
+				Doc:         "URL to fetch data from. Supported schemas are `http` and `https`",
+			},
+			&dataspec.AttrSpec{
+				Name:       "method",
+				Type:       cty.String,
+				DefaultVal: cty.StringVal("GET"),
+				OneOf: []cty.Value{
+					cty.StringVal("GET"),
+					cty.StringVal("POST"),
+					cty.StringVal("HEAD"),
+				},
+				Doc: "HTTP method for the request. Allowed methods are `GET`, `POST` and `HEAD`",
+			},
+			&dataspec.AttrSpec{
+				Name:       "basic_auth_username",
+				Type:       cty.String,
+				DefaultVal: cty.NullVal(cty.String),
+				Doc:        `A basic authentication username`,
+			},
+			&dataspec.AttrSpec{
+				Name:       "basic_auth_password",
+				Type:       cty.String,
+				Secret:     true,
+				DefaultVal: cty.NullVal(cty.String),
+				Doc:        `A basic authentication password`,
+			},
+			&dataspec.AttrSpec{
+				Name:       "insecure",
+				Type:       cty.Bool,
+				DefaultVal: cty.BoolVal(false),
+				Doc:        "If set to `true`, disabled verification of the server's certificate.",
+			},
+			&dataspec.AttrSpec{
+				Name:        "timeout_secs",
```

Comment:
I only have this tidbit to add:
Perhaps this can be a [go duration](https://pkg.go.dev/maze.io/x/duration#ParseDuration) string?

Reply:
this number is converted into a duration value during processing. Do you have something else in mind?

Reply:
@traut standard `time` package have `ParseDuration` that @Andrew-Morozko  linked here. It takes `string` and converts it to `time.Duration`. You can just use string values like "5m" or "15s". So no need for `_secs` suffix then.

Reply:
aha! will take a look! 🙇

Reply:
fixed! thank you both!